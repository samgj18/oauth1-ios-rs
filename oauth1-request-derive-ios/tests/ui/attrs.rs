#[derive(oauth1_request_ios::Request)]
#[oauth1(unknown)]
struct UnknownWord {}

#[derive(oauth1_request_ios::Request)]
#[oauth1(unknown::path)]
struct UnknownPathWord {}

#[derive(oauth1_request_ios::Request)]
#[oauth1(unknown = "")]
struct UnknownNameValue {}

#[derive(oauth1_request_ios::Request)]
#[oauth1(unknown::path = "")]
struct UnknownPathName {}

#[derive(oauth1_request_ios::Request)]
#[oauth1(unknown::path(""))]
struct UnknownList {}

#[derive(oauth1_request_ios::Request)]
#[oauth1(crate)]
struct ExpectedNameValueUnexpectedWord {}

#[derive(oauth1_request_ios::Request)]
#[oauth1(crate(""))]
struct ExpectedNameValueUnexpectedList {}

#[derive(oauth1_request_ios::Request)]
#[oauth1(crate = oauth1_request)]
#[oauth1(crate = oauth1_request)]
struct DuplicateNameValue {}

#[derive(oauth1_request_ios::Request)]
#[oauth1(crate = oauth1_request, crate = oauth1_request)]
struct DuplicateNameValue2 {}

#[derive(oauth1_request_ios::Request)]
struct Fields {
    #[oauth1(rename = 0)]
    non_str_lit: u8,

    #[oauth1(skip = "")]
    expected_word_unexpected_name_value: u8,

    #[oauth1(skip(""))]
    expected_word_unexpected_list: u8,

    #[oauth1(skip_if)]
    expected_name_value_unexpected_word: u8,

    #[oauth1(skip_if(""))]
    expected_name_value_unexpected_list: u8,

    #[oauth1(unknown)]
    unknown_word: u8,

    #[oauth1(unknown::path)]
    unknown_path_word: u8,

    #[oauth1(unknown = "")]
    unknown_name_value: u8,

    #[oauth1(unknown::path = "")]
    unknown_path_name: u8,

    #[oauth1(unknown(""))]
    unknown_list: u8,

    #[oauth1(unknown::path(""))]
    unknown_path_list_name: u8,

    #[oauth1(skip, skip)]
    duplicate_word: u8,

    #[oauth1(encoded)]
    #[oauth1(encoded)]
    duplicate_word_2: u8,

    #[oauth1(rename = "a", rename = "b")]
    duplicate_name_value: u8,

    #[oauth1(rename = "c")]
    #[oauth1(rename = "d")]
    duplicate_name_value_2: u8,

    #[oauth1(option = FALSE)]
    invalid_boolean: u8,

    #[oauth1(fmt = |_, _| Ok(()))]
    invalid_path: u8,

    duplicate: u8,
    #[oauth1(rename = "duplicate")]
    duplicate_renamed: u8,

    #[oauth1(rename = "?")]
    uri_unsafe: u8,
}

fn main() {}
