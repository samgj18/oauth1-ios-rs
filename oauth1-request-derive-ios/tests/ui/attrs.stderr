error: unknown attribute `unknown`
 --> tests/ui/attrs.rs:2:10
  |
2 | #[oauth1(unknown)]
  |          ^^^^^^^

error: unknown attribute `unknown::path`
 --> tests/ui/attrs.rs:6:10
  |
6 | #[oauth1(unknown::path)]
  |          ^^^^^^^

error: unknown attribute `unknown`
  --> tests/ui/attrs.rs:10:10
   |
10 | #[oauth1(unknown = "")]
   |          ^^^^^^^

error: unknown attribute `unknown::path`
  --> tests/ui/attrs.rs:14:10
   |
14 | #[oauth1(unknown::path = "")]
   |          ^^^^^^^

error: unknown attribute `unknown::path`
  --> tests/ui/attrs.rs:18:10
   |
18 | #[oauth1(unknown::path(""))]
   |          ^^^^^^^

error: expected name-value meta
  --> tests/ui/attrs.rs:22:10
   |
22 | #[oauth1(crate)]
   |          ^^^^^

error: expected name-value meta
  --> tests/ui/attrs.rs:26:10
   |
26 | #[oauth1(crate(""))]
   |          ^^^^^

error: duplicate attribute `crate`
  --> tests/ui/attrs.rs:31:10
   |
31 | #[oauth1(crate = oauth1_request)]
   |          ^^^^^

error: duplicate attribute `crate`
  --> tests/ui/attrs.rs:35:34
   |
35 | #[oauth1(crate = oauth1_request, crate = oauth1_request)]
   |                                  ^^^^^

error: expected string literal
  --> tests/ui/attrs.rs:40:23
   |
40 |     #[oauth1(rename = 0)]
   |                       ^

error: expected meta word
  --> tests/ui/attrs.rs:43:14
   |
43 |     #[oauth1(skip = "")]
   |              ^^^^

error: expected meta word
  --> tests/ui/attrs.rs:46:14
   |
46 |     #[oauth1(skip(""))]
   |              ^^^^

error: expected name-value meta
  --> tests/ui/attrs.rs:49:14
   |
49 |     #[oauth1(skip_if)]
   |              ^^^^^^^

error: expected name-value meta
  --> tests/ui/attrs.rs:52:14
   |
52 |     #[oauth1(skip_if(""))]
   |              ^^^^^^^

error: unknown attribute `unknown`
  --> tests/ui/attrs.rs:55:14
   |
55 |     #[oauth1(unknown)]
   |              ^^^^^^^

error: unknown attribute `unknown::path`
  --> tests/ui/attrs.rs:58:14
   |
58 |     #[oauth1(unknown::path)]
   |              ^^^^^^^

error: unknown attribute `unknown`
  --> tests/ui/attrs.rs:61:14
   |
61 |     #[oauth1(unknown = "")]
   |              ^^^^^^^

error: unknown attribute `unknown::path`
  --> tests/ui/attrs.rs:64:14
   |
64 |     #[oauth1(unknown::path = "")]
   |              ^^^^^^^

error: unknown attribute `unknown`
  --> tests/ui/attrs.rs:67:14
   |
67 |     #[oauth1(unknown(""))]
   |              ^^^^^^^

error: unknown attribute `unknown::path`
  --> tests/ui/attrs.rs:70:14
   |
70 |     #[oauth1(unknown::path(""))]
   |              ^^^^^^^

error: duplicate attribute `skip`
  --> tests/ui/attrs.rs:73:20
   |
73 |     #[oauth1(skip, skip)]
   |                    ^^^^

error: duplicate attribute `encoded`
  --> tests/ui/attrs.rs:77:14
   |
77 |     #[oauth1(encoded)]
   |              ^^^^^^^

error: duplicate attribute `rename`
  --> tests/ui/attrs.rs:80:28
   |
80 |     #[oauth1(rename = "a", rename = "b")]
   |                            ^^^^^^

error: duplicate attribute `rename`
  --> tests/ui/attrs.rs:84:14
   |
84 |     #[oauth1(rename = "d")]
   |              ^^^^^^

error: expected boolean literal
  --> tests/ui/attrs.rs:87:23
   |
87 |     #[oauth1(option = FALSE)]
   |                       ^^^^^

error: expected path
  --> tests/ui/attrs.rs:90:20
   |
90 |     #[oauth1(fmt = |_, _| Ok(()))]
   |                    ^

error: parameter name must be URI-safe
  --> tests/ui/attrs.rs:97:23
   |
97 |     #[oauth1(rename = "?")]
   |                       ^^^

error: duplicate parameter "duplicate"
  --> tests/ui/attrs.rs:94:23
   |
94 |     #[oauth1(rename = "duplicate")]
   |                       ^^^^^^^^^^^
